manageiq-operator/pkg/controller/manageiq/manageiq_controller.go

```diff
@@ -267,6 +267,12 @@ func (r *ReconcileManageIQ) generateMemcachedResources(cr *miqv1alpha1.ManageIQ)
 }
 
 func (r *ReconcileManageIQ) generatePostgresqlResources(cr *miqv1alpha1.ManageIQ) error {
+	hostName := getSecretKeyValue(r.client, cr.Namespace, cr.Spec.DatabaseSecret, "hostname")
+	if hostName != "" {
```

Comment:
@abellotti Should this be `if hostName != "postgresql" {`?

Reply:
No, this is what triggers external vs internal - https://github.com/ManageIQ/manageiq-pods/pull/604#issuecomment-670514247

Reply:
It's not obvious to me why this would be empty string since either the user creates the secret with a hostname or we do... https://github.com/ManageIQ/manageiq-pods/blob/89efdbe14731a17b4eb33777725d9562aa5fac44/manageiq-operator/pkg/helpers/miq-components/postgresql.go#L19

Reply:
It should stay if hostName != "", if the user specified  a secret with a non-blank hostname we're done, we don't get to call that DefaultPostgresqlSecret (where we populate the secret for the local postgresql case), which is called on line 276 after the if clause.

Reply:
@bdunne Yeah I had the same question - I assumed when we generate it ourselves we don't specify a hostname?